README.Rmd

```diff
@@ -46,28 +46,113 @@ forcats is now part of the core tidyverse, so you do not need to load it explici
 library(tidyverse)
 ```
 
-Factors are used to describe categorical variables with a fixed and known set of __levels__. You can create factors with the base `factor()` or [`readr::parse_factor()`](http://readr.tidyverse.org/reference/parse_factor.html):
+Let's try answering the question "what are the most common homeworld of starwars characters?" Let's start off by making a barplot
+
+```{r initial_plot}
+ggplot(starwars, aes(x = hair_color)) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+That's okay, but it would be more helpful the graph was ordered by count. This is a case of an **unordered** categorical variable where we want it ordered by its frequency. We can use the function `fct_infreq`: 
+
+```{r fct_infreq_hair}
+ggplot(starwars, aes(x = fct_infreq(hair_color))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+Note that it automatically puts NA at the top, even though that doesn't have the smallest number of entries. 
+
+What if we want to do a similar plot for skin color? 
+
+```{r fct_infreq_skin}
+ggplot(starwars, aes(x = fct_infreq(skin_color))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+Here we have a lot of different skin colors - let's reduce it to only be the top 5. We can use `fct_lump` to "lump" all the infrequent colors into one factor, "other." The argument `n` is the number of levels we want to keep.
+
+```{r fct_lump}
+ggplot(starwars, aes(x = fct_lump(skin_color, n = 5))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+We could also have used `prop` instead, which keeps all the levels that appear at least `prop` of the time. For example, let's keep skin colors that at least 10% of the characters have: 
+
+```{r fct_lump_prop}
+ggplot(starwars, aes(x = fct_lump(skin_color, prop = .1))) + 
+  geom_bar() + 
+  coord_flip()
+```
+
+Only light and fair remain; everything else is other. 
+
+What if we wanted to see if the average mass differed by eye color? We'll only look at the 6 most popular eye colors and remove NAs.  
+
+```{r fct_lump_mean}
+starwars %>%
+  mutate(eye_color = fct_lump(eye_color, n = 6)) %>%
+  group_by(eye_color) %>%
+  summarise(mean_mass = mean(mass, na.rm = TRUE)) %>%
+  ggplot(aes(x = eye_color, y = mean_mass)) + 
+  geom_col()
+```
+
+It looks like people (or at least one person) with orange eyes is definitely heavier! This graph would look nice though if it was ordered by `mean_mass`. We can do this with `fct_reorder`, which reorders one variable by another. 
```

Comment:
* I think it's people…are, since "one person" is in parens.
* parens for `fct_reorder()` ( just adding another set of eyes for these, since they're easy to miss with args, etc. in the mix)!